File: src/main/java/cz/spse/nahradnihodnoceni/models/data/Exam.java
package cz.spse.nahradnihodnoceni.models.data;

import jakarta.persistence.*;
import lombok.*;

import java.util.Date;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Getter
@Setter
public class Exam {

    public static final String[] marks = {"1","2","3","4","5","N"};

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Builder.Default
    private boolean available = true;

    @ManyToOne(optional = false)
    private Student student;

    @ManyToOne(optional = false)
    private Subject subject;

    @ManyToOne
    private Classroom classroom;

    @ManyToOne
    private Teacher chairman;

    @ManyToOne
    private Teacher classTeacher;

    @ManyToOne(optional = false)
    private Teacher examiner;

    @Temporal(TemporalType.TIMESTAMP)
    private Date time;

    private int year;
    private int period;

    @Column(nullable = false)
    private String originalMark;

    private String finalMark;
}

File: frontend/src/models/data/Exam.ts
import Classroom from './Classroom';
import Student from './Student';
import Subject from './Subject';
import Teacher from './Teacher';

export type FinalMarkType = '1' | '2' | '3' | '4' | '5' | null;

interface Exam {
  id: number;
  available: boolean;

  student: Student;
  subject: Subject;
  classroom: Classroom | null;
  chairman: Teacher | null;
  class_teacher: Teacher | null;
  examiner: Teacher;
  time: number | null;
  year: number;
  period: number;
  originalMark: '5' | 'N';
  finalMark: FinalMarkType;
}

export default Exam;
